### PR TITLE
Use a Python virtual environment

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request updates the Docker image configuration to use a Python virtual environment.

> [!NOTE]
> I am creating this pull request against a branch that will represent a larger body of work. This branch will later have a pull request against `develop` once the body of work is completed.

## 💭 Motivation and context ##

Using virtual environments is a Python best practice. It will also ensure our Python dependencies are isolated from the system Python environment.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -42,25 +43,32 @@ RUN addgroup --system --gid ${CISA_GID} ${CISA_GROUP} \
...
 #
-# Note that we use pip3 --no-cache-dir to avoid writing to a local
-# cache.  This results in a smaller final image, at the cost of
-# slightly longer install times.
+# Note that we sym-link the Python binary in the venv to the system-wide Python so that
+# any calls to `python3` will use our virtual environment. We are using short flags
+# because the ln binary in Alpine Linux does not support long flags.
 ###
-RUN pip3 install --no-cache-dir https://github.com/cisagov/skeleton-python-library/archive/v${VERSION}.tar.gz
+RUN ln -sf "$(command -v python3)" "${VIRTUAL_ENV}"/bin/python3
```
Comment (jsf9k) [inline Dockerfile]: Nice trick!

Please add a short comment explaining what the `-s` and `-f` flags do since we cannot use long options here.

**Review by jsf9k — CHANGES_REQUESTED** [review]
One minor ask, otherwise this looks great!

File: Dockerfile (hunk 1 above)
Comment (mcdonnnj) [reply Dockerfile]: Please let me know if [77b5e34](https://github.com/cisagov/skeleton-docker/pull/190/commits/77b5e3483d2063358c20fe5b04648a9c777a8bdd) works for you.

**Review by dav3r — APPROVED** [review]
Very nice! 👍

**Review by jsf9k — APPROVED** [review]

**Pull request merged**